### PR TITLE
[internal-dns] remove lookup_ipv6 in favor of lookup_socket_v6

### PR DESCRIPTION
**Pull request by iximeow**
`lookup_ipv6` is both buggy and easy to misuse:
* it sends an AAAA query for a domain which should have a SRV record
  - this works only because https://github.com/oxidecomputer/omicron/issues/4051 means the SRV record is incorrectly returned, along with the actually-desired AAAA for the SRV's target in Additionals
* it looks up an IPv6 address from a SRV record *but ignores the port*. in places `lookup_ipv6` was used, it was paired consistently with the hardcoded port NEXUS_INTERNAL_PORT and matched what should be in the resolved SRV record. if we for example wanted to move Nexus' port (or start a test Nexus on an atypical port), the authoritative port number in the SRV response would be ignored for the hardcoded port.

lets just use the port that we told DNS we're at!

we may still want a bare IPv6 address for a service if we're going to test network reachability, for example, but we're not doing that with this function today. this all is distinct from helpers like `lookup_all_ipv6`.

if we need a service's IPv6 address to use with an alternate port to access a different API, we *probably* should have a distinct SRV record for that lookup to use instead? i've found three instances of this:
* [wicket](https://github.com/oxidecomputer/omicron/blob/d25b102/wicketd/src/nexus_proxy.rs#L122-L141) assumes that the techport proxy can be done on the same IP as Nexus' API, just a different port
* we assume the [CRDB admin service](https://github.com/oxidecomputer/omicron/blob/d25b102/nexus/reconfigurator/execution/src/omicron_zones.rs#L194-L200) listens on the same IP as CRDB itself, but i don't think that *has* to be true?
* we look up addresses for MGS [here](https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/bootstrap/early_networking.rs#L262) via `ServiceName::Dendrite`, but there's a `ServiceName::ManagementGatewayService` that - if it doesn't already - seems like it should have a SRV record too.

there are some uses of `lookup_all_ipv6` that make a lot of sense still, where we're discovering the rack's network and _really_ do not care about the port that Dendrite happens to be on.

(this all said, `lookup_ipv6` would be simple enough to fix and retain. but after looking at its uses it seems it might be more trouble than it's worth right now)

this is the counterpart to https://github.com/oxidecomputer/omicron/pull/6308 which fixes clients (so far as the test suite and i can tell) to not depend on our internal DNS server returning answers we didn't ask for. we could land this first and #6308 at some point later, or i _think_ since Oximeter is the only place we depended on `lookup_ipv6`, i'm pretty sure we can land both?

### CODE REVIEW

File: oximeter/collector/src/agent.rs [hunk 1]
```diff
@@ -827,12 +826,10 @@ async fn resolve_nexus_with_backoff(
...
     let do_lookup = || async {
         resolver
-            .lookup_ipv6(ServiceName::Nexus)
+            .lookup_socket_v6(ServiceName::Nexus)
             .await
             .map_err(|e| BackoffError::transient(e.to_string()))
-            .map(|ip| {
-                SocketAddr::V6(SocketAddrV6::new(ip, NEXUS_INTERNAL_PORT, 0, 0))
-            })
+            .map(|addr| SocketAddr::V6(addr))
```
Comment (luqmana) [inline oximeter/collector/src/agent.rs:832]: Maybe worth adding a `lookup_socket` that returns a `SocketAddr`?

**Review by luqmana — APPROVED** [review]
> * it looks up an IPv6 address from a SRV record _but ignores the port_. in places `lookup_ipv6` was used, it was paired consistently with the hardcoded port NEXUS_INTERNAL_PORT and matched what should be in the resolved SRV record. if we for example wanted to move Nexus' port (or start a test Nexus on an atypical port), the authoritative port number in the SRV response would be ignored for the hardcoded port.
> 
> 
> lets just use the port that we told DNS we're at!

Yea, that's always been the goal so glad to see this!

Dunno about the CRDB admin service case, but for the other two:

> * [wicket](https://github.com/oxidecomputer/omicron/blob/d25b102/wicketd/src/nexus_proxy.rs#L122-L141) assumes that the techport proxy can be done on the same IP as Nexus' API, just a different port

Ah yea, the dropshot instance that serves the external api for the techport proxy listens on the underlay network by binding to the same address as the internal nexus api. An explicit SRV/service entry for it though wouldn't be a bad idea to get rid of hardcoding ports everywhere.

> * we look up addresses for MGS [here](https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/bootstrap/early_networking.rs#L262) via `ServiceName::Dendrite`, but there's a `ServiceName::ManagementGatewayService` that - if it doesn't already - seems like it should have a SRV record too.

Yea, I think that's just a typo. Looking up `ManagementGatewayService` should work.

PR looks good with one suggestion. The specific service fixes can be done as a follow up.

File: oximeter/collector/src/agent.rs (hunk 1 above)
Comment (iximeow) [reply oximeter/collector/src/agent.rs:832]: almost tempted, looked at other uses of `lookup_socket_v6` and it turns out they do almost the same thing as we're about to do here: format the `SocketAddrV6` into something like `http://{}`. so instead i'll make this match other use sites and keep it as `SocketAddrV6` the whole way.

(i'd like `lookup_socket` a lot, but it'd be backed by `lookup_service_targets` which is specifically ipv6-only. so the implication that `lookup_socket` could work for SRV records referencing ipv4 targets would be misleading)

**Review by luqmana — APPROVED** [review]

**Comment by iximeow** [comment]
OK. this should actually pass (finally). failures were not because of changes at any point here, they were because a combination of https://github.com/oxidecomputer/omicron/pull/6204, this branch being made before #6204, and the illumos used here by buildomat being upgraded to include the change that motivated #6204. so entirely incidentally to this PR, `ipadm` was not erroring in exactly the way we expected it would to hint that we needed to create the addresses, and instead everything got wedged.

merging main pulls in #6204 and with a :heavy_check_mark: i'll merge.

**Pull request merged**